doc/nrf/ug_thingy91.rst

```diff
@@ -0,0 +1,262 @@
+=================================================
+
+When you have the pre-compiled firmware images ready, you can directly program the images onto the Thingy:91 using the `nRF Connect Programmer`_ app, which is implemented as part of the `nRF Connect for Desktop`_.
+
+This is a direct USB method, making use of the MCUboot feature and the inbuilt serial recovery mode of Thingy:91.
+You can program the Thingy:91 either on the nRF9160 SiP or on the nRF52840 SoC component.
+
+For the detailed procedure for programming a Thingy:91 using nRF Connect Programmer, see the following: (links to the thingy91 programming sections in nRF connect programmer userguide in infocenter to be added here).
+
+Updating modem firmware
+=======================
+
+You can update the modem firmware of Thingy:91 by using an external debug probe such as nRF9160 DK or any J-Link device supporting ARM Cortex-M33.
+See (link to the thingy91 modem update section in nRF connect programmer userguide in infocenter to be added here) for the detailed steps to update the modem firmware.
+
+.. _building_pgming:
+
+Building and programming from the source code
+*********************************************
+
+You can also program the Thingy:91 by using the images obtained by building the code in an |NCS| environment.
+
+To set up your system to be able to build a compatible firmware image, follow the `nRF Connect SDK Getting started guide`_.
+The build targets of interest for  Thingy:91 in |NCS|, are as follows:
+
++---------------+--------------------------------------------------+
+|Component      |  Build Target                                    |
++===============+==================================================+
+|nRF9160 SiP    |``nrf9160_pca20035`` for the secure version       |
+|               |                                                  |
+|               |``nrf9160_pca20035ns`` for the non-secure version |
++---------------+--------------------------------------------------+
+|nRF52840 SoC   |``nrf52840_pca20035``                             |
++---------------+--------------------------------------------------+
+
+You should use the build target ``nrf9160_pca20035`` or ``nrf9160_pca20035ns``, when building the application code for the nRF9160 SiP and the build target ``nrf52840_pca20035``, when building the application code for the onboard nRF52840 SoC.
+
+.. note::
+
+   LTE/GPS features can only be used with non-secure target.
+
+The table below shows the different types of build files that are generated and the different scenarios in which they are used:
+
++--------------+----------------------------------------+---------------------------------------+
+| File         | File format                            | Programming scenario                  |
++==============+========================================+=======================================+
+|merged.hex    | Full image, HEX format                 | Using debug probe                     |
```

Comment:
Using external debug probe and nRF Connect Programmer

Reply:
Modified.